omniqueue/src/backends/azure_queue_storage.rs

```diff
@@ -0,0 +1,196 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use azure_storage::StorageCredentials;
+use azure_storage_queues::{
+    operations::Message, PopReceipt, QueueClient, QueueServiceClientBuilder,
+};
+use serde::Serialize;
+
+use crate::{queue::Acker, Delivery, QueueBackend, QueueError, Result};
+
+fn get_client(cfg: &AqsQueueConfig) -> QueueClient {
+    let storage_credentials =
+        StorageCredentials::access_key(cfg.storage_account.clone(), cfg.access_key.clone());
+
+    let mut builder =
+        QueueServiceClientBuilder::new(cfg.storage_account.clone(), storage_credentials);
+    if let Some(cloud_uri) = cfg.cloud_uri.clone() {
+        builder = builder.cloud_location(azure_storage::CloudLocation::Custom {
+            account: cfg.storage_account.clone(),
+            uri: cloud_uri,
+        });
+    }
+    builder.build().queue_client(cfg.queue_name.clone())
+}
+
+pub struct AqsQueueBackend;
```

Comment:
Sorry for another nitpick, but I realized that this is needlessly verbose, I removed `Queue` from backend type names a while ago plus the `Q` in `AQS` stands for queue anyways. Also I don't think this users should be able to instantiate this struct.

```suggestion
#[non_exhaustive]
pub struct AqsBackend;
```